cfgov/v1/models/browse_page.py

```diff
@@ -72,11 +76,31 @@ class BrowsePage(CFGOVPage):
 
     objects = PageManager()
 
+    def get_mortgage_meta(self):
+        from data_research.models import MortgageMetaData
+        meta_set = MortgageMetaData.objects.all()
+        meta = {obj.name: json.loads(obj.json_value) for obj in meta_set}
+        meta['thru_date'] = meta['sampling_dates'][-1]
+        meta['thru_date_formatted'] = parser.parse(
+            meta['thru_date']).strftime("%B %-d, %Y")
+        meta_sample = meta.get(
+            'download_files')[meta['thru_date']]['percent_90']['County']
+        meta['pub_date'] = meta_sample['pub_date']
+        meta['pub_date_formatted'] = parser.parse(
+            meta['pub_date']).strftime("%B %-d, %Y")
+        return meta
+
     def add_page_js(self, js):
         super(BrowsePage, self).add_page_js(js)
         js['template'] += ['secondary-navigation.js']
 
     def get_context(self, request, *args, **kwargs):
         context = super(BrowsePage, self).get_context(request, *args, **kwargs)
         context.update({'get_secondary_nav_items': get_secondary_nav_items})
+        if 'data-research/mortgage' in request.url:
+            context.update(self.get_mortgage_meta())
+        if '30-89' in request.url:
+            context.update({'delinquency': 'percent_30_60'})
+        elif '90' in request.url:
+            context.update({'delinquency': 'percent_90'})
```

Comment:
I understand the rationale but worry that doing this kind of URL-based logic is error-prone and goes a bit too far in violating the Wagtail model. This means that, for purposes of unit testing or local development of pages with these blocks, users/developers have to just somehow know the right hardcoded path to use. I know we do this thing elsewhere but it would be better if blocks/pages could somehow be self-describing/encapsulate their own behavior based on some consistent logic.

Is "no new page types" a hard and fast rule? Tagging @Scotchester as I know he has run into this kind of thing before, wanting custom behavior on a page based at least in part on its location in the tree. Is there some other distinguishing factor we could use, i.e. a page category?